internal/app/secretless/handlers/mysql/connection_details_test.go

```diff
@@ -0,0 +1,83 @@
+package mysql
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConnectionDetailsCorrectlyParsesConnectionDetails(t *testing.T) {
+	credentials := make(map[string][]byte)
+
+	credentials["host"] = []byte("myhost")
+	credentials["port"] = []byte("1234")
+	credentials["username"] = []byte("myusername")
+	credentials["password"] = []byte("mypassword")
+
+	expectedConnDetails := ConnectionDetails{
+		Host:     "myhost",
+		Port:     1234,
+		Username: "myusername",
+		Password: "mypassword",
+		Options:  make(map[string]string),
+	}
+
+	actualConnDetails, err := NewConnectionDetails(credentials)
+	assert.Nil(t, err)
+
+	assert.EqualValues(t, expectedConnDetails, *actualConnDetails)
+}
+
+func TestPortDefaultsTo3306IfNotProvided(t *testing.T) {
+	credentials := make(map[string][]byte)
+
+	credentials["host"] = []byte("myhost")
+	credentials["username"] = []byte("myusername")
+	credentials["password"] = []byte("mypassword")
+
+	expectedConnDetails := ConnectionDetails{
+		Host:     "myhost",
+		Port:     3306,
```

Comment:
Similar to my other comment, this [magic number](https://refactoring.guru/replace-magic-number-with-symbolic-constant) should refer to a public constant on the impl file, rather than be hardcoded in both places.